src/frontend/screens/Invites/InviteReceived.tsx

```diff
@@ -0,0 +1,168 @@
+import {BottomSheetWrapper} from '../../sharedComponents/BottomSheetWrapper';
+import InviteIcon from '../../images/AddPersonCircle.svg';
+import {StyleSheet, View} from 'react-native';
+import {LIGHT_GREY} from '../../lib/styles';
+import {PrimaryButton, SecondaryButton} from '../../sharedComponents/Buttons';
+import {defineMessages, useIntl} from 'react-intl';
+import {NativeNavigationComponent} from '../../sharedTypes/navigation';
+import {
+  useAcceptInvite,
+  useClientApi,
+  useRejectInvite,
+  useSingleInvite,
+} from '@comapeo/core-react';
+import {useEffect} from 'react';
+import {HeaderText} from '../../sharedComponents/Text/HeaderText';
+import {BodyText} from '../../sharedComponents/Text/BodyText';
+import {UIActivityIndicator} from 'react-native-indicators';
+import {useActiveProjectIdActions} from '../../contexts/ActiveProjectIdStoreContext';
+import * as Sentry from '@sentry/react-native';
+import {Invite} from '@comapeo/core/dist/invite/invite-api';
+
+const m = defineMessages({
+  navTitle: {
+    id: 'screens.InviteReceived.navTitle',
+    defaultMessage: 'Invite Received',
+  },
+  acceptInvite: {
+    id: 'screens.InviteReceived.acceptInvite',
+    defaultMessage: 'Accept Invite',
+  },
+  declineInvite: {
+    id: 'screens.InviteReceived.declineInvite',
+    defaultMessage: 'Decline Invite',
+  },
+  joinProject: {
+    id: 'screens.InviteReceived.joinProject',
+    defaultMessage: 'Join {projectName}',
+  },
+  invitedToJoin: {
+    id: 'screens.InviteReceived.invitedToJoin',
+    defaultMessage: "You've been invited to join {projectName}",
+  },
+});
+
+export const InviteReceived: NativeNavigationComponent<'InviteReceived'> = ({
+  route,
+  navigation,
+}) => {
+  const {formatMessage} = useIntl();
+  const inviteId = route.params.inviteId;
+  const {data: invite} = useSingleInvite({inviteId});
+  const acceptInvite = useAcceptInvite();
+  const rejectInvite = useRejectInvite();
+  const {setActiveProjectId} = useActiveProjectIdActions();
+  const {invite: mapeoApiInvite} = useClientApi();
+
+  useEffect(() => {
+    function navigateBasedOnInviteState(invite: Invite) {
+      if (invite.inviteId !== inviteId) return;
+
+      if (invite.state === 'canceled') {
+        navigation.replace('InviteCancelled', {
+          projectName: invite.projectName,
+        });
+        return;
+      }
+
+      if (invite.state !== 'pending') {
+        navigation.goBack();
+      }
+    }
+
+    mapeoApiInvite.addListener('invite-updated', navigateBasedOnInviteState);
+
+    return () => {
+      mapeoApiInvite.removeListener(
+        'invite-updated',
+        navigateBasedOnInviteState,
+      );
+    };
+  }, [mapeoApiInvite, inviteId, navigation]);
```

Comment:
this is the most significant change. Now we are listening for updated to the invite state, and navigating accordingly. More specifically, if the invite gets cancelled, the user is navigated to the cancel screen. Any other state that is not `pending` the user is navigated away from this screen (in order to avoid the user getting stuck on this screen)

Reply:
I don't know if I am understanding your question fully. But the invite itself doesn't actually expose an error. It has an `error` state, but it doesn't have a message. We catch errors (and send them to sentry) where there is an error accepting or rejecting an invite.

Reply:
an invite with `state: 'error'` also has an `error: Error` field (see https://github.com/digidem/comapeo-core/blob/6d88d68cf2d505fb6c6d8af2d4b73a45a2a10682/src/invite/invite-api.js#L35)

Reply:
oh interesting! We dont ever expose errors to the user. I will send the error to sentry though

Reply:
Reporting to sentry could suffice then. If we get errors during the next training would it be possible to get the sentry logs for those devices?

Reply:
https://github.com/digidem/comapeo-mobile/pull/1076/commits/3cbd895e41b2c86f119f2ebf10f0bc7d9759f869 and https://github.com/digidem/comapeo-mobile/pull/1076/commits/9feabb133f647644317070546a8b535673b2dfb4
So now we are sending that error to sentry and popping up the error bottom sheet. Im not 100% sure if we need to show the error bottom sheet. Like we don't ever expose the errors to user, so it might just be the same to clear the invite. Perhaps that is a question for design

Reply:
> Reporting to sentry could suffice then. If we get errors during the next training would it be possible to get the sentry logs for those devices?

yes sentry logs all the device info along with the errors